apps/backend/src/logging/logging.service.ts

```diff
@@ -0,0 +1,170 @@
+        format: 'MMM-DD-YYYY HH:mm:ss'
+      }),
+      winston.format.printf(
+        (info) => `[${[info.timestamp]} ${info.ip}] ${info.message}`
+      )
+    )
+  });
+
+  // Actions involving evaluations
+  logEvaluationAction(
+    request: {user: User},
+    ip: string,
+    action: string,
+    evaluation: EvaluationDto
+  ): void {
+    this.logger.info({
+      ip: ip,
+      message: `${this.userToString(
+        request.user
+      )}: ${action}: ${this.evaluationToString(evaluation)}`
+    });
+  }
+
+  evaluationToString(evaluation: Evaluation | EvaluationDto): string {
+    return `Evaluation<ID: ${evaluation.id} | Filename: ${evaluation.filename}>`;
+  }
+
+  // Actions involving evaluation tags
+  logEvaluationTagAction(
+    request: {user: User},
+    ip: string,
+    action: string,
+    tag: EvaluationTagDto
+  ): void {
+    this.logger.info({
+      ip: ip,
+      message: `${this.userToString(
+        request.user
+      )}: ${action}: ${this.evaluationTagToString(tag)}`
+    });
+  }
+
+  evaluationTagToString(tag: EvaluationTagDto): string {
+    return `EvaluationTag<ID: ${tag.id} | Value: ${tag.value}>`;
+  }
+
+  evaluationTagsToString(tags: EvaluationTagDto[]): string {
+    return `EvaluationTags[${tags
+      .map((tag) => this.evaluationTagToString(tag))
+      .join(', ')}]`;
+  }
+
+  // Actions involving groups
+  logGroupAction(
+    request: {user: User},
+    ip: string,
+    action: string,
+    group: GroupDto
+  ): void {
+    this.logger.info({
+      ip: ip,
+      message: `${this.userToString(
+        request.user
+      )}: ${action}: ${this.groupToStringWithUsers(group)}`
+    });
+  }
+
+  logGroupActionWithEvaluationSubject(
+    request: {user: User},
+    ip: string,
+    action: string,
+    subject: Evaluation | EvaluationDto,
+    group: GroupDto
+  ): void {
+    this.logger.info({
+      ip: ip,
+      message: `${this.userToString(
+        request.user
+      )}: ${action} ${this.evaluationToString(subject)} -> ${this.groupToString(
+        group
+      )}`
+    });
+  }
+
+  logGroupActionWithUserSubject(
+    request: {user: User},
+    ip: string,
+    action: string,
+    subject: User,
+    group: GroupDto
+  ): void {
+    this.logger.info({
+      ip: ip,
+      message: `${this.userToString(
+        request.user
+      )}: ${action} ${this.userToString(subject)} -> ${this.groupToString(
+        group
+      )}`
+    });
+  }
+
+  groupToString(group: GroupDto): string {
+    return `Group<ID: ${group.id} | Name: ${group.name}>`;
+  }
+
+  groupToStringWithUsers(group: GroupDto): string {
+    return `Group<ID: ${group.id} | Name: ${group.name} | Users[${group.users
+      .map((user) => this.userToString(user))
+      .join(', ')}]>`;
+  }
+
+  // Actions involving authentication
+  logAuthenticationAction(request: Request, action: string, user?: User): void {
+    this.logger.info({
+      ip: request.ip,
+      message: `${action}: ${this.userToString(user)}`
```

Comment:
```suggestion
      message: `${request.originalUrl} ${request.method} ${action}: ${this.userToString(user)}`
```

Will produce `[Jun-24-2021 14:01:16 ::ffff:127.0.0.1] /authn/login POST Succeded Login As: User<ID: 1>`